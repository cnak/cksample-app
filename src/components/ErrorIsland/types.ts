export interface ValidationError {
    field: string;
    message: string;
}

export interface ErrorWithPress extends ValidationError {
    onPress?: () => void;
}

export interface ErrorIslandProps {
    errors?: Array<ErrorWithPress>;
}
