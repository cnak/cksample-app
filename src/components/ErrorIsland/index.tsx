import React from 'react';
import styled from 'styled-components/native';

import { H2, rem } from '../Typography';
import { ERROR_COLOUR } from '../colors';

import { ErrorIslandProps } from './types';

const StyledView = styled.View`
    border: 4px solid ${ERROR_COLOUR};
    padding: 15px;
    margin-bottom: 15px;
`;

export default ({ errors }: ErrorIslandProps) => {
    if (!errors || errors.length === 0) {
        return null;
    }

    return (
        <StyledView>
            <H2 style={{ fontSize: rem(1.125), marginBottom: 0 }} accessibilityLiveRegion="assertive">
                There is a problem
            </H2>
        </StyledView>
    );
};
