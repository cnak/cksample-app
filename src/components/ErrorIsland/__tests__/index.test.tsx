import React from 'react';
import renderer from 'react-test-renderer';

import ErrorIsland from '..';

describe('<ErrorIsland>', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    it('matches snapshot', () => {
        const errors = [{ field: 'test', message: 'Test error message' }];
        const tree = renderer.create(<ErrorIsland errors={errors}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
