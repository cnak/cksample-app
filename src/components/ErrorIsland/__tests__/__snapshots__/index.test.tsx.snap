// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorIsland> matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#B10E1E",
        "borderStyle": "solid",
        "borderWidth": 4,
        "marginBottom": 15,
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
      },
    ]
  }
>
  <Text
    accessibilityLiveRegion="assertive"
    style={
      Array [
        Object {
          "color": "#0B0C0C",
          "fontFamily": "Helvetica",
          "fontSize": 24,
          "fontWeight": "bold",
          "lineHeight": 25.000080000000004,
          "marginBottom": 15,
        },
        Object {
          "fontSize": 18,
          "marginBottom": 0,
        },
      ]
    }
  >
    There is a problem
  </Text>
</View>
`;
